src/utils/testHelpers.ts

```diff
@@ -82,9 +82,21 @@ export function mockAPICalls(testName: string, url = MainnetUrl) {
  * or the one with attached INFURA SECRET key from
  * UNSTOPPABLE_RESOLUTION_INFURA_PROJECTID env variable if any
  */
-export function secretInfuraLink(): string {
+export function secretInfuraLink(options: InfuraProtocolOptions = InfuraProtocolOptions.http): string {
   const secret = process.env.UNSTOPPABLE_RESOLUTION_INFURA_PROJECTID;
-  let url = 'https://mainnet.infura.io';
-  if (secret) url = `https://mainnet.infura.io/v3/${secret}`;
-  return url;
+  let url = "";
+  switch (options) {
```

Comment:
Is there any better-looking way to work with enums beside the switch case? This looks too big in my opinion for such a small change.

Reply:
Make a map like:

```
const url = {wss: 'url1', http: 'url2'}[options]
```

Reply:
So simple and elegant! Added this way. Strange I haven't thought of that before! Thanks!